src/shipchain_common/aws.py

```diff
@@ -0,0 +1,137 @@
+"""
+Copyright 2020 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+import logging
+
+import requests
+from aws_requests_auth.boto_utils import BotoAWSRequestsAuth
+from django.conf import settings
+from influxdb_metrics.loader import log_metric, TimingMetric
+from rest_framework import status
+
+from .exceptions import AWSIoTError
+from .utils import DecimalEncoder
+
+LOG = logging.getLogger('python-common')
+
+
+class AWSClient:
```

Comment:
The way the current refactor was implemented, `AWSClient` appears to be an incomplete abstract class.  It expects there to be a few properties defined on the subclasses that will be instantiated, but there is no assertions or assurances that those properties must be defined before we start to use them.

If this is instantiated as-is (without a subclass) it will not function.  This class either needs to be made in to a proper abstract class with properties that must be defined in the subclasses or it should be reworked further to allow the passing of required values to each called method with checks that each variable has been defined or provided to the methods before we use them.

Reply:
The AWSClient was intended to be more of a mixin than it's own class, as I didn't think a default url or authentication would be accurate with how the refactor intended to make it abstract.